packages/fields/src/types/Slug/Implementation.js

```diff
@@ -199,72 +200,43 @@ export class SlugImplementation extends Text {
       return slug;
     }
 
-    let uniqueSlug = slug;
-    let slugIsUnique;
-    let makeUniqueAttempts = 0;
     const listAndFieldPath = `${this.listKey}.${this.path}`;
 
-    // The "all<List>s" query
-    const { listQueryMetaName, whereInputName } = this.getListByKey(this.listKey).gqlNames;
-
-    // A query to find any _other_ items with the same slug
-    const queryString = `
-      query findDuplicate($where: ${whereInputName}) {
-        ${listQueryMetaName}(where: $where) {
-          count
-        }
-      }
-    `;
-
-    if (this.alwaysMakeUnique) {
-      // Force a uniquification pass over the slug
-      uniqueSlug = await this.makeUnique({ slug, previousSlug: uniqueSlug });
-    }
-
     // Repeat until we have a unique slug, or we've tried too many times
-    do {
-      if (makeUniqueAttempts >= MAX_UNIQUE_ATTEMPTS) {
-        throw new Error(
-          `Attempted to generate a unique slug for ${listAndFieldPath}, but failed after too many attempts. If you've passed a custom 'makeUnique' function, ensure it is working correctly`
-        );
+    let uniqueSlug = slug;
+    for (let i = 0; i < MAX_UNIQUE_ATTEMPTS; i += 1) {
+      if (this.alwaysMakeUnique || i > 0) {
+        uniqueSlug = await this.makeUnique({ slug, previousSlug: uniqueSlug });
       }
-      makeUniqueAttempts++;
 
-      // Check to see if the slug is unique
-      const { data, errors } = await context.executeGraphQL({
-        // Access Control may filter out some results, so we wouldn't be
-        // retreiving an accurate list of all existing items. Because we add the
-        // unique constraint to the field, the database will throw an error if
-        // we miss a match and try to insert anyway.
-        context: context.createContext({ skipAccessControl: true }),
-        query: queryString,
-        variables: {
+      try {
+        const result = await getItems({
```

Comment:
IMO we can use some helper in utils to help simplify queries if needed. (I see the query is dumped in favour of getItems)

Reply:
`getItem` *is* the query helper, there's no need to re-invent that wheel. This is the exact use case that `getItem` was designed to support, e.g. running queries from within a hook.